### DISCUSSION TITLE
React and Vite - Why is still loading other component not imported

### DISCUSSION THREAD

**Discussion by FastRunDev** [community/community#58885] [Programming Help] [answered]
### Body

I have made some configs on tsconfig.paths.json and this is how I structure files below.

My problem is with the importing, Is it good to import just like this?

import { Products } from '@components'.

My problem is because I notice it loads all components even though I didn't import all. For example if I go to /sample-page/, it loads People, Food, Events.

Would it better if I do import Products from '@components/Products', so it won't load other components such as People, Food, Events?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by iamsuiux** [comment, marked as answer]
When you do this import: import { Products } from '@components', you're actually importing from the index.ts file in the @components directory (this is a default behavior in JavaScript/TypeScript). This index.ts file, in turn, is importing all your components (Products, People, Food, Events). So, even if you're only using Products in SamplePage, all components get loaded because they're all imported in index.ts.

You're correct that changing the import to import Products from '@components/Products' would solve this issue. This way, you're only importing the Products component directly, and the others (People, Food, Events) would not be loaded unless they're also directly imported somewhere in your code.

So, your intuition is right. If you want to ensure that only the components you're directly using are loaded, it's better to import them directly from their own files, instead of from an index.ts file that imports everything
